engine/src/main/java/org/terasology/engine/core/subsystem/lwjgl/LwjglGraphics.java

```diff
@@ -108,14 +118,22 @@ public void preShutdown() {
                 int[] xBuffer = new int[1];
                 int[] yBuffer = new int[1];
                 GLFW.glfwGetWindowPos(window, xBuffer, yBuffer);
-                config.setWindowPosX(xBuffer[0]);
-                config.setWindowPosY(yBuffer[0]);
 
                 int[] widthBuffer = new int[1];
                 int[] heightBuffer = new int[1];
                 GLFW.glfwGetWindowSize(window, widthBuffer, heightBuffer);
-                config.setWindowWidth(widthBuffer[0]);
-                config.setWindowHeight(heightBuffer[0]);
+                if (widthBuffer[0]<= 0 && heightBuffer[0]<=0){
+                    config.setWindowPosX(defaultX);
+                    config.setWindowPosY(defaultY);
+                    config.setWindowWidth(defaultWidth);
+                    config.setWindowHeight(defaultHeight);
```

Comment:
If we have no valid data for this, what if we skip these `config.setWindow*` calls entirely in that case? Would they keep their old values?

Other suggestion: make the the conditional on zero width **or** height, not **and**.

Reply:
Yes, that would keep the old values(Default as in #4160). But I suggest keeping it because it gives the user to adjust their screen size and position as they want and keep it so they do not want to re-adjust every time they open the game.  :)

Reply:
I'm suggesting we **do** write the values when we _have_ valid values, but skip it in the `0` case.

Reply:
I don't understand defaultX and defaultY for window position.

Reply:
oh ok.... you are suggesting 
`if (widthBuffer[0]>0 && heightBuffer[0]>0 && xBuffer[0]>0 && yBuffer[0]>0){
                    config.setWindowWidth(widthBuffer[0]);
                    config.setWindowHeight(heightBuffer[0]);
                    config.setWindowPosX(xBuffer[0]);
                    config.setWindowPosY(yBuffer[0]);
                }`

Yes this is better. I tried it and working fine too.  :)
So we don't need default x and y values.